Fix: Action/Detail/Tree spacing with hidden columns

ingvaldlorentzen opened the pull request:
## Related Issue
#1121 

## Description
Fixes a bug that would cause the iconbutton (action/detail/tree etc) on the right side to take on the "extra" space "leftover" when hiding columns. See the related issue for how this looks.   
This is fixed by filtering out hidden columns when checking the length for the CSS `calc`. 

## Related PRs
None that I could find.

## Impacted Areas in Application
* data-manager.js

gaurav-agarwal27 commented:
@mbrn how long it can take to merge and close this PR ? Really need this

Shmew commented:
I ran into this problem earlier this week. You can get around this in the mean time by doing a theme override of the `.MuiTableCell-paddingNone` and `.MuiTableCell-paddingCheckbox` by setting the width to `calc((100% - 0px) / 16)`

raethlein commented:
I stumbled on the same issue today. Great to see that a PR is already open! 

I would slightly modify the code and apply the hidden test already in line 61: 
` const undefinedWidthColumns = columns.filter(c => c.width === undefined && !c.hidden)`
as there filtering is already applied.

simondriesen commented:
> I ran into this problem earlier this week. You can get around this in the mean time by doing a theme override of the `.MuiTableCell-paddingNone` and `.MuiTableCell-paddingCheckbox` by setting the width to `calc((100% - 0px) / 16)`

If you are using the detailPanel prop this solution will break the detailPanel button width. This theme override fixed it for me:

```
MuiTableCell: {
  alignLeft: {
    width: 'auto !important',
  },
},
```

ingvaldlorentzen commented:
> Could you remove the changes to white space, so that the changes that this PR is actually doing are more visible?
Fixed.
> Also since the filter is already applied in line 61, could you move the filter up to that and see if that works too?
Good catch! I moved it to the existing `filter()`, works well in all my environments!
